pkg/loki/loki.go

```diff
@@ -571,22 +571,27 @@ func (t *Loki) setupModuleManager() error {
 	}
 
 	if t.Cfg.InternalServer.Enable {
-		depsToUpdate := []string{Distributor, Ingester, Querier, QueryFrontendTripperware, QueryScheduler, Ruler, TableManager, Compactor, IndexGateway}
-
-		for _, dep := range depsToUpdate {
-			var idx int
-			for i, v := range deps[dep] {
+		for key, ds := range deps {
+			idx := -1
+			for i, v := range ds {
 				if v == Server {
 					idx = i
 					break
 				}
 			}
 
-			lhs := deps[dep][0 : idx+1]
-			rhs := deps[dep][idx+2:]
+			if idx == -1 {
+				continue
+			}
 
-			deps[dep] = append(lhs, InternalServer)
-			deps[dep] = append(deps[dep], rhs...)
+			if len(ds) == idx {
```

Comment:
I don't think this clause is possible. The only thing setting `idx` is the range over `ds` so it could never find an index at `len(ds)`, no?

Is this supposed to be `if len(ds)-1 == idx` to indicate a shortcut if the server is the last dependency index? If so, I don't think that optimization is needed 🤔